corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js

```diff
@@ -54,22 +54,6 @@ hqDefine("cloudcare/js/formplayer/menus/collections", function () {
         parse: function (response) {
             _.extend(this, _.pick(response, this.commonProperties));
 
-            Util.doUrlAction((urlObject) => {
-                let updateUrl = false;
-                if (!urlObject.appId && response.appId) {
-                // will be undefined on urlObject when coming from an incomplete form
-                    urlObject.appId = response.appId;
-                    this.appId = urlObject.appId;
-                    updateUrl = true;
-                }
-                if (response.selections) {
-                    urlObject.setSelections(response.selections);
-                    sessionStorage.removeItem('selectedValues');
```

Comment:
@stephherbers FYI this moves the logic you're updating in https://github.com/dimagi/commcare-hq/pull/31949/